components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx

```diff
@@ -378,13 +403,24 @@ function ConfirmTransactionPanel ({
           buttonType='reject'
           text={getLocale('braveWalletAllowSpendRejectButton')}
           onSubmit={onReject}
+          disabled={transactionConfirmed}
         />
-        <NavButton
-          buttonType='confirm'
-          text={getLocale('braveWalletAllowSpendConfirmButton')}
-          onSubmit={onConfirm}
-          disabled={isConfirmButtonDisabled}
-        />
+        {transactionConfirmed ? (
+          <ConfirmingButton>
+            <ConfirmingButtonText>
+              {getLocale('braveWalletAllowSpendConfirmButton')}
+            </ConfirmingButtonText>
+            <LoadIcon />
+          </ConfirmingButton>
+        ) : (
+          <NavButton
+            buttonType='confirm'
+            text={getLocale('braveWalletAllowSpendConfirmButton')}
+            onSubmit={onClickConfirmTransaction}
+            disabled={isConfirmButtonDisabled || transactionConfirmed}
```

Comment:
`transactionConfirmed` will never be true in this case. can be removed

Reply:
Removed!